mindsdb/interfaces/model/model_controller.py

```diff
@@ -0,0 +1,262 @@
+        from mindsdb.interfaces.model.learn_process import LearnProcess, run_learn
+
+        join_learn_process = kwargs.get('join_learn_process', False)
+        if 'join_learn_process' in kwargs:
+            del kwargs['join_learn_process']
+
+        self._setup_for_creation(name)
+
+        if ray_based:
+            run_learn(name, from_data, to_predict, kwargs, datasource_id)
+        else:
+            p = LearnProcess(name, from_data, to_predict, kwargs, datasource_id)
+            p.start()
+            if join_learn_process is True:
+                p.join()
+                if p.exitcode != 0:
+                    raise Exception('Learning process failed !')
+        return 0
+
+    def predict(self, name, pred_format, when_data=None, kwargs={}):
+        from mindsdb_datasources import FileDS, ClickhouseDS, MariaDS, MySqlDS, PostgresDS, MSSQLDS, MongoDS, SnowflakeDS, AthenaDS
+        import mindsdb_native
+        from mindsdb.interfaces.storage.db import session, Predictor
+        if name not in self.predictor_cache:
+            # Clear the cache entirely if we have less than 1.2 GB left
+            if psutil.virtual_memory().available < 1.2 * pow(10,9):
+                self.predictor_cache = {}
+
+            predictor_record = Predictor.query.filter_by(company_id=self.company_id, name=name, is_custom=False).first()
+            if predictor_record.data['status'] == 'complete':
+                self.fs_store.get(name, f'predictor_{self.company_id}_{predictor_record.id}', self.config['paths']['predictors'])
+                self.predictor_cache[name] = {
+                    'predictor': mindsdb_native.Predictor(name=name, run_env={'trigger': 'mindsdb'}),
+                    'created': datetime.datetime.now()
+                }
+
+        if isinstance(when_data, dict) and 'kwargs' in when_data and 'args' in when_data:
+            data_source = getattr(mindsdb_datasources, when_data['class'])(*when_data['args'], **when_data['kwargs'])
+        else:
+            # @TODO: Replace with Datasource
+            try:
+                data_source = pd.DataFrame(when_data)
+            except Exception as e:
+                data_source = when_data
+
+        predictions = self.predictor_cache[name]['predictor'].predict(
+            when_data=when_data,
+            **kwargs
+        )
+        if pred_format == 'explain' or pred_format == 'new_explain':
+            predictions = [p.explain() for p in predictions]
+        elif pred_format == 'dict':
+            predictions = [p.as_dict() for p in predictions]
+        elif pred_format == 'dict&explain':
+            predictions = [[p.as_dict() for p in predictions], [p.explain() for p in predictions]]
+        else:
+            raise Exception(f'Unkown predictions format: {pred_format}')
+
+        if ray_based:
+            return predictions
+        return xmlrpc.client.Binary(pickle.dumps(predictions))
```

Comment:
may be add to class
```
def _return(self, obj):
        if ray_based:
            return obj
        return xmlrpc.client.Binary(pickle.dumps(obj))
```
and change returns to call to that function? just to make it slightly more beauty :)